src-ts/fs-promised.ts

```diff
@@ -0,0 +1,86 @@
+/**
+ * Promise and async iterator friendly filesystem APIs.
+ */
+
+import { Flags } from "uv";
+import * as fs from "./fs-uv.js";
+import { AsyncDataStream, iterateBody } from "./utils2.js";
+
+/** Open a file. */
+export async function open(path: string, flags: Flags, mode: number): Promise<number> {
+    return new Promise((resolve, reject) =>
+        fs.open(path, flags, mode, (error, fd) =>
+            error ? reject(error) : resolve(fd)));
+}
+
+export async function close(fd: number): Promise<void> {
+    return new Promise((resolve, reject) =>
+        fs.close(fd, (error) =>
+            error ? reject(error) : resolve()));
+}
+
+export async function read(fd: number, buffer: Uint8Array, position: number): Promise<number> {
+    return new Promise((resolve, reject) =>
+        fs.read(fd, buffer, position, (error, bytesRead) =>
```

Comment:
what will happen if a developer calls this without providing position?

Reply:
Probably a type error in the C++ conversion code or a libuv error code.

This isn't a public API, it's an implementation detail.